Content.Server/Fluids/EntitySystems/AbsorbentSystem.cs

```diff
@@ -158,11 +159,18 @@ private bool TryTransferAbsorber(EntityUid user, EntityUid used, EntityUid targe
             return false;
         }
 
-        absorberSoln.AddReagent(PuddleSystem.EvaporationReagent, water);
-        refillableSolution.AddSolution(nonWater, _prototype);
 
-        _solutionSystem.UpdateChemicals(used, absorberSoln);
-        _solutionSystem.UpdateChemicals(target, refillableSolution);
+        if (water > 0 && !_solutionContainerSystem.TryAddReagent(used, absorberSoln, PuddleSystem.EvaporationReagent, water,
```

Comment:
Can't you use TryAddSolution for this one too?

Reply:
Solution tries to add whole thing inside and if fails doesn't put any, where reagent will add as much as it can and if some won't fit wil return false.